### ISSUE TITLE
Wrong line and line number shown for certain errors when run through 6to5-node

### ISSUE THREAD

**Issue by atombender** [6to5/6to5#658]
Example use case:

```javascript
#!node_modules/.bin/6to5-node

require("6to5/register")({
  //ignore: false
});

import express from 'express';

var Foo = 3;

var app = express();

new Foo();
```

This reports:

```
/tmp/testcase/bad.js:7
import express from 'express';
                             ^
TypeError: number is not a function
    at Object.<anonymous> (/tmp/testcase/bad.js:7:30)
    at Module._compile (module.js:456:26)
    at Object.loader (/tmp/testcase/node_modules/6to5/lib/6to5/register.js:73:5)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at Object.<anonymous> (/tmp/testcase/node_modules/6to5/bin/_6to5-node:92:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
```

The real error obviously occurs on line 13.

This test case has been reduced from a larger program. If I change it just a little bit, for example by removing the commented-out line 4, or the very first shebang, it works fine. Not sure what's triggering it, all I can say is that the above test case does it.

Transpiling with `6to5` also produced the correct result.

Versions:

* 6to5 3.3.2
* Node 0.10.33.

**Comment by sebmck** [comment]
Looks like this is actually a result of the shebang as initially I tried just:

```javascript
require("6to5/register")({
  //ignore: false
});

import express from 'express';

var Foo = 3;

var app = express();

new Foo();
```

and it worked perfectly. This should be a quick fix.

**Comment by atombender** [comment]
I'm getting the error without a shebang, too. This is as far as I've been able to whittle it down:

```javascript
'use strict';

// Enables ES6 mode. Run app with 6to5-node for server-side goodness.
require("6to5/register")({
  //ignore: false
});

import express from 'express';

var Foo = 3;

var app = express();
app.use(express.static(__dirname + '/public'));
app.listen(3000);

new Foo();
```

**Comment by atombender** [comment]
Even smaller:

```javascript:
'use strict';

require("6to5/register")({
  //ignore: false
});

var Foo = 3;

import fs from 'fs';

new Foo();
```

**Comment by sebmck** [comment]
Interesting, I'll look into it when I can. Hopefully it's nothing too serious as the source maps and location tracking has been pretty solid so far.

**Comment by bjoerge** [comment]
Its probably because the require hook gets registered twice in the above examples. When you use `6to5-node`, the require hook is registered for you, so in the code examples above it gets registered two times. This, in turn causes already transpiled code to be transpiled twice. If you remove `require("6to5/register)(...)` it should work (at least it does for me).

@sebmck would it be a good idea to guard against these situations? I.e. check if the require hook has already been registered and if so either skip or warn?

**Comment by sebmck** [comment]
@bjoerge It shouldn't be. The register hook has the ability to be ran multiple times. The location tracking magically fixes itself when you remove the `"use strict";` statement too so it's highly unlikely to be the cause.

**Comment by bjoerge** [comment]
Ah, nevermind then.

**Comment by sebmck** [comment]
@bjoerge Thanks for looking it though, appreciated!

**Comment by bjoerge** [comment]
No problem, happy to (try to) help out!

Managed to boil it down to an even smaller example:

`bad.js`:
```js
"use strict"
var Foo = 3;
new Foo();
```

```$ ./node_modules/.bin/6to5-node bad.js```

Reports line 1, but expected line 3:
```
/tmp/bad.js:1
"use strict"
            ^
TypeError: number is not a function
    at Object.<anonymous> (/tmp/bad.js:1:13)
    at Module._compile (module.js:456:26)
    at Object.loader (/tmp/node_modules/6to5/lib/6to5/register.js:73:5)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at Object.<anonymous> (/tmp/node_modules/6to5/bin/_6to5-node:92:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
```
```
$ node_modules/.bin/6to5-node --version
$ 3.3.3
```
And yeah, like you said removing "use strict" seems to correct it.

**Comment by sebmck** [comment]
Fixed as of 3.3.4, thanks!

**Comment by atombender** [comment]
Terrific. Thank _you_!
